syntax = "proto3";

import "map_renderer.proto";
import "graph.proto";
import "transport_router.proto";

package tc_serialize;

message Stop{
	int32 id = 1;
	string name = 2;
	double lat = 3;
	double lng = 4;
}

message Bus{
	int32 id = 1;
	string name = 2;
	repeated int32 route = 3;
	bool is_round = 4;
	repeated int32 half_route = 5;
}

message Distance{
	int32 first_stop_id = 1;
	int32 second_stop_id = 2;
	int32 distance = 3;
}

message TransportBase{
	repeated Stop stops = 1;
	repeated Bus buses = 2;
	repeated Distance distances = 3;
	RenderSettings rend_set = 4;
	Graph graph = 5;
	Router router = 6;
	TransportRouter transport_router = 7;
}

	